helix-term/src/application.rs

```diff
@@ -276,8 +276,21 @@ impl Application {
                     debugger.is_running = false;
 
                     // whichever thread stops is made "current" (if no previously selected thread).
-                    if let Some(thread_id) = thread_id {
-                        select_thread_id(&mut self.editor, thread_id, false).await;
+                    if thread_id.is_none() || all_threads_stopped.unwrap_or_default() {
```

Comment:
I think we should avoid this hack altogether, none of the other clients do any special handling if `thread_id` is `nil` and instead expect you to manually select a thread (`(lsp--error "Currently active thread is not stopped. Use 'dap-switch-thread' or select stopped thread from sessions view.")`)

https://github.com/emacs-lsp/dap-mode/blob/8ec7e98986ea46e3c36b7ecbdf9a6562c90c7632/dap-mode.el#L942

https://github.com/mfussenegger/nvim-dap/blob/980bdf8e68d626adba0fea7ef990966bd47052a1/lua/dap/session.lua#L269-L275

Reply:
I think it would be convenient to have main selected automatically.

Reply:
It will auto select it once it stops on a breakpoint.